Makefile

```diff
@@ -434,3 +434,11 @@ verify-helm: gen-helm ## Check Helm charts are up to date
 	@echo 'If this test fails, it is because the git diff is non-empty after running "make gen-helm".'
 	@echo 'To correct this, locally run "make gen-helm", commit the changes, and re-run tests.'
 	git diff --quiet --exit-code charts
+
+##################################
+# HELP
+##################################
+
+.PHONY: help
+help: ## Shows the available commands
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```

Comment:
Would be good if can we add `##` target comment in other useful targets to be shown in `make help` command.

Reply:
I agree, can we do that progressively ? Or does it need to be all done in this PR ?

Reply:
Not all of them , some of them which are used in day to day bases, like `verify-codegen` or similar

Reply:
Here is the output of `make help`:
```
gen-crd-api-reference-docs     Install gen-crd-api-reference-docs
generate-api-docs              Generate api reference docs
verify-api-docs                Check api reference docs are up to date
kustomize                      Install kustomize
kustomize-crd                  Create install.yaml
codegen                        Update all generated code and docs
verify-api                     Check api is up to date
verify-config                  Check config is up to date
verify-codegen                 Verify all generated code and docs are up to date
gen-helm-docs                  Generate Helm docs
gen-helm                       Generate Helm charts stuff
verify-helm                    Check Helm charts are up to date
help                           Shows the available commands
```

I think it covers the most used ones but let me know if there are some missing important ones and I will add them to this PR.

Reply:
IMO above 2 should be good enough for now. further we can do progressively  wdyt ?

Reply:
looks good , thanks

Reply:
I think this is a good start, other ones can come later IMHO.

Reply:
for me the output is : missing `verify-codegen , codegen` targets

```
make help
gen-crd-api-reference-docs     Install gen-crd-api-reference-docs
generate-api-docs              Generate api reference docs
verify-api-docs                Check api reference docs are up to date
kustomize                      Install kustomize
kustomize-crd                  Create install.yaml
gen-helm-docs                  Generate Helm docs
gen-helm                       Generate Helm charts stuff
verify-helm                    Check Helm charts are up to date
check-helm-docs                Check Helm docs
help                           Shows the available commands

```

Reply:
`git pull` ? ;-)

Reply:
aah right , works fine now.